### ISSUE TITLE
The valid-jsdoc rule doesn't catch missing param for multi-line assignment expressions

### ISSUE THREAD

**Issue by tschaub** [eslint/eslint#4889]
With ESLint @ 1972678ba03c25d1e5babff17c2d2559116a4886, if an assignment expression spans multiple lines, `valid-jsdoc` doesn't catch missing `@param` or param description.  See the following example:

```js
/* eslint valid-jsdoc: [2, {requireReturn: false}] */

var obj = {};

/**
 * Single line assignment.  The valid-jsdoc rule catches the issue with the
 * missing param description.
 * @param {string} arg
 * @return {string} A string.
 */
obj.foo = function(arg) {
  return arg;
};

/**
 * Multi line assignment.  The valid-jsdoc rule misses the issue with the
 * missing param description.
 * @param {string} arg
 * @return {string} A string.
 */
obj.bar =
    function(arg) {
      return arg;
    };
```
>   5:1  error  Missing JSDoc parameter description for 'arg'  valid-jsdoc

The first comment block is correctly linted (it is missing the parameter description).  The second comment block is not correctly linted (it is also missing the parameter description).

**Comment by eslintbot** [comment]
@tschaub Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by alberto** [comment]
I guess the other cases will fail too: "ClassExpression" and "FunctionDeclaration". We should probably fix them here too.

**Comment by gyandeeps** [comment]
I have fixed them also. Will add tests for it.
